analyzer/src/main/kotlin/managers/Maven.kt

```diff
@@ -66,13 +66,10 @@ class Maven(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfigu
     private inner class LocalProjectWorkspaceReader : WorkspaceReader {
         private val workspaceRepository = WorkspaceRepository()
 
-        override fun findArtifact(artifact: Artifact): File? {
-            return if (artifact.extension == "pom") {
-                localProjectBuildingResults[artifact.identifier()]?.pomFile?.absoluteFile
-            } else {
-                null
-            }
-        }
+        override fun findArtifact(artifact: Artifact) =
+                localProjectBuildingResults[artifact.identifier()]?.pomFile?.absoluteFile.takeIf {
+                    artifact.extension == "pom"
+                }
```

Comment:
Using `takeIf` without using `it` inside seems like a misuse of the function. Also, it moves the check here, so there will be unnecessary map lookups. What about using a helper function for such cases:

```kotlin
override fun findArtifact(artifact: Artifact) =
    ifOrNull(artifact.extension == "pom") {
        localProjectBuildingResults[artifact.identifier()]?.pomFile?.absoluteFile
    }

inline fun <T> ifOrNull(condition: Boolean, block: () -> T) = if (condition) block() else null
```

Reply:
Good point about the potentially unnecessary map lookup. I've reverted the `takeIf` changes as I don't like to introduce yet another helper function just for this case.

Reply:
I just realized there is a way without another helper by moving the `takeIf`:

```kotlin
override fun findArtifact(artifact: Artifact) =
    artifact.takeIf { it.extension == "pom" }?.let {
        localProjectBuildingResults[it.identifier()]?.pomFile?.absoluteFile
    }
```

I'll approve and leave it up to you if you want to change it again.

Reply:
Good idea, I'll add a new commit.